beacon-chain/sync/initial-sync/round_robin.go

```diff
@@ -126,18 +126,35 @@ func (s *InitialSync) roundRobinSync(genesis time.Time) error {
 			return blocks[i].Slot < blocks[j].Slot
 		})
 
-		for _, blk := range blocks {
-			logSyncStatus(genesis, blk, peers, counter)
-			if featureconfig.FeatureConfig().InitSyncNoVerify {
-				if err := s.chain.ReceiveBlockNoVerify(ctx, blk); err != nil {
+		var receiveBlocks func(blocks []*eth.BeaconBlock) error
+		receiveBlocks = func(blocks []*eth.BeaconBlock) error {
```

Comment:
this is dangerous, what if we never find a parent that exists in our db ?

Reply:
also this method should also be a pointer receiver

Reply:
i can limit the depth of it. but it should satisfy both successfully retrieving a parent forever

Reply:
changed to pointer receiver

Reply:
limited the depth of recursive calls to 10